frontend/src/features/documents/hooks/useDocumentRelationshipProvider.ts

```diff
@@ -110,12 +112,43 @@ export const useDocumentRelationshipProvider = () => {
     }, []),
   });
 
+  // Provides functionality for uploading a document for a given relationship
+  const { execute: updateDocument, loading: updateDocumentLoading } = useApiRequestWrapper<
+    (
+      relationshipType: DocumentRelationshipType,
+      documentId: number,
+      updateRequest: Api_DocumentUpdateRequest,
+    ) => Promise<AxiosResponse<boolean, any>>
+  >({
+    requestFunction: useCallback(
+      async (
+        relationshipType: DocumentRelationshipType,
+        documentId: number,
+        updateRequest: Api_DocumentUpdateRequest,
+      ) => await updateDocumentMetadataApiCall(relationshipType, documentId, updateRequest),
+      [updateDocumentMetadataApiCall],
+    ),
+    requestName: 'updateDocumentMetadataApiCall',
+    onSuccess: useCallback(() => {
+      toast.success('Updated document metadata');
+    }, []),
+    onError: useCallback((axiosError: AxiosError<IApiError>) => {
```

Comment:
have a look at useAxiosSucccessHandler and useAxiosErrorHandler to cut down on this boilerplate.

Reply:
we are using useApiRequestWrapper react hook  where internally axios response error are handled.